src/Leaves/Pager.php

```diff
@@ -144,10 +137,15 @@ protected function beforeRender()
             $this->model->numberOfPages = $this->calculateNumberOfPages();
         }
 
+        // If we will be unable to present the page as it's outside our bounds, we 
+        // go back to page 1.
+        if ($this->model->pageNumber > $this->model->numberOfPages){
+            $this->setPageNumber(1);
+        }
```

Comment:
Just want to confirm, this no longer throws the `PagerOutOfBoundsException`?

Reply:
Yeah, we wait now instead until render time when we shift to page 1 without an exception.

There is a small wrinkle here - if the pager is beneath the table or in other contexts calculated after your iterations, you maybe end up with a mismatch with the data presented and the page number indicated by the pager. However I can't think of any situation like that and this is too big an optimisation to miss.